smithy-cli/src/main/java/software/amazon/smithy/cli/commands/InitCommand.java

```diff
@@ -165,6 +272,8 @@ public void registerHelp(HelpPrinter printer) {
                     "Smithy templates repository url");
             printer.param("--output", "-o", "new-smithy-project",
                     "Smithy project directory");
+            printer.param("--list", "-l", null,
```

Comment:
Given that this is a new mode of calling `smithy init`, it would be nice to provide example commands like what I added in the `smithy format` PR.

```
Examples:

smithy init --list
smithy init -o /tmp/quickstart-gradle -t quickstart-gradle